src/NuGet.Clients/PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -102,64 +113,67 @@ public NuGetProject DefaultNuGetProject
                 throw new ArgumentNullException(nameof(projectSystemFactory));
             }
 
+            if (logger == null)
+            {
+                throw new ArgumentNullException(nameof(logger));
+            }
+
+            _serviceProvider = serviceProvider;
             _projectSystemCache = projectSystemCache;
             _projectSystemFactory = projectSystemFactory;
+            _logger = logger;
+        }
 
-            _dte = ServiceLocator.GetInstance<DTE>();
-            _vsSolution = ServiceLocator.GetGlobalService<SVsSolution, IVsSolution>();
-            _vsMonitorSelection = ServiceLocator.GetGlobalService<SVsShellMonitorSelection, IVsMonitorSelection>();
+        public async Task InitializeAsync(IAsyncServiceProvider site)
+        {
```

Comment:
why do you need to move all the initialization code to separate API? Now the problem is solutionManager is constructed but still not initialized which could lead to problems for cases like template wizard api calls where SolutionRestoreManager won't be in picture.
So I'll suggest either keep initialization code inside constructor or call this initialization api from constructor.